packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -201,6 +201,29 @@ export default class CashierStore extends BaseStore {
         this.onSwitchAccount(this.accountSwitcherListener);
     }
 
+    // Initial init just for displaying P2P banner without mounting the entire cashier
+    @action.bound
+    init() {
+        // eslint-disable-next-line no-undef
+        reaction(
+            () => [this.root_store.client.is_logged_in, this.root_store.client.residence],
+            () => {
+                if (!this.is_p2p_visible && !this.root_store.client.is_virtual) {
+                    WS.authorized.p2pAdvertiserInfo().then(advertiser_info => {
```

Comment:
can we call this once and set is p2p visible with it as well instead of calling it twice, once on load of app and once on load of cashier? either getting the response from cache in onmountcommon or setting isp2pvisible here as well

Reply:
Will refactor it into only calling it on app load instead, and not on cashier load.